File: springboot-version/backend/src/main/java/com/petconnect/entity/MissingPetContact.java
package com.petconnect.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.LocalDateTime;

@Entity
@Table(name = "missing_pet_contacts")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MissingPetContact {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "report_id", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private MissingPetReport report;

    @ManyToOne
    @JoinColumn(name = "contact_user_id", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private User contactUser;

    @Column(columnDefinition = "TEXT")
    private String message;

    private String contactPhone;
    private String contactEmail;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: aspnet-version/backend/PetConnectAlmost/Models/MissingPetContact.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PetConnectAlmost.Models
{
    [Table("missing_pet_contacts")]
    public class MissingPetContact
    {
        [Key]
        [Column("id")]
        public long Id { get; set; }

        [Column("report_id")]
        public long ReportId { get; set; }

        [Column("contact_user_id")]
        public long ContactUserId { get; set; }

        [Column("message")]
        public string? Message { get; set; }

        [Column("contact_phone")]
        public string? ContactPhone { get; set; }

        [Column("contact_email")]
        public string? ContactEmail { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; }

        // Navigation
        [ForeignKey("ReportId")]
        public virtual MissingPetReport Report { get; set; }

        [ForeignKey("ContactUserId")]
        public virtual User ContactUser { get; set; }
    }
}
